src/collections/match/operations/match-getall-me.operation.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  BlapiResponse,
+  BlError,
+  Match,
+  MatchVariant,
+  StandMatch,
+  UserDetail,
+  UserMatch,
+} from "@boklisten/bl-model";
+import { BlDocumentStorage } from "../../../storage/blDocumentStorage";
+import { matchSchema } from "../match.schema";
+import { BlCollectionName } from "../../bl-collection";
+import { Operation } from "../../../operation/operation";
+import { BlApiRequest } from "../../../request/bl-api-request";
+import { userDetailSchema } from "../../user-detail/user-detail.schema";
+import { SEDbQuery } from "../../../query/se.db-query";
+import { ObjectId } from "mongodb";
+import { User } from "../../user/user";
+import { UserSchema } from "../../user/user.schema";
+
+// TODO: Move these two to bl-model
+interface RelevantDetails {
+  name: string;
+}
+type MatchWithDetails =
+  | StandMatch
+  | (UserMatch & {
+      senderDetails: RelevantDetails;
+      receiverDetails: RelevantDetails;
+    });
+
+export class GetMyMatchesOperation implements Operation {
+  constructor(
+    private userStorage?: BlDocumentStorage<User>,
+    private userDetailStorage?: BlDocumentStorage<UserDetail>,
+    private matchStorage?: BlDocumentStorage<Match>
+  ) {
+    this.userStorage = userStorage
+      ? userStorage
+      : new BlDocumentStorage(BlCollectionName.Users, UserSchema);
+    this.userDetailStorage = userDetailStorage
+      ? userDetailStorage
+      : new BlDocumentStorage(BlCollectionName.UserDetails, userDetailSchema);
+    this.matchStorage = matchStorage
+      ? matchStorage
+      : new BlDocumentStorage(BlCollectionName.Matches, matchSchema);
+  }
+
+  async run(blApiRequest: BlApiRequest): Promise<BlapiResponse> {
+    const meQuery = new SEDbQuery();
+    meQuery.stringFilters = [
+      {
+        fieldName: "blid",
+        value: blApiRequest.user.id,
```

Comment:
Why not use the primary key for userDetail?

```suggestion
        fieldName: "_id",
        value: blApiRequest.user.id,
```

Reply:
@AdrianAndersen

Reply:
It certainly does include the userDetail primary key. 

However, it seems like the API `handleRequest` (higher order method) did not forward it to `blApiRequest` for some reason. That is stupid, so I changed it.😛

Reply:
Ah ok, I only looked at the actual properties in the debugger